datalad/distribution/install.py

```diff
@@ -270,13 +308,37 @@ def __call__(dataset=None, path=None, source=None, recursive=False,
         # if we have no dataset given, figure out which one we need to operate
         # on, based on the resolved target location (that is now guaranteed to
         # be specified
-        if ds is None:
+        if ds is None and path is not None:
             # try to find a dataset at or above the installation target
             dspath = GitRepo.get_toppath(abspath(path))
             if dspath is None:
                 # no top-level dataset found, use path as such
                 dspath = path
             ds = Dataset(dspath)
+
+        if ds is None and path is None and source is not None:
+            # we got nothing but a source. do something similar to git clone
+            # and derive the path from the source and continue
+            lgr.debug(
+                "Neither dataset not target installation path provided. "
+                "Assuming installation of a remote dataset. "
+                "Deriving destination path from given source {0}".format(
+                    source))
+            dspath = source.rstrip('/').split('/')
+            if dspath[-1] == '.git':
+                dspath = dspath[-2]
+            else:
+                dspath = dspath[-1]
+            if dspath.endswith('.git'):
+                dspath = dspath[:-4]
+            ds = Dataset(dspath)
+
+        if not path and ds is None:
+                # no dataset, no target location, nothing to do
```

Comment:
Actually, no I won't. The repeated `ds is None` tests have to be sequential, as a `ds` could emerge inbetween.